tests/cypress/support/commands.js

```diff
@@ -13,73 +13,77 @@ require('../plugins/compareImages/compareImagesCommand');
 let selectedValueGlobal = '';
 
 Cypress.Commands.add('login', (username = Cypress.env('user'), password = Cypress.env('password')) => {
-    cy.get('[placeholder="Username"]').type(username);
-    cy.get('[placeholder="Password"]').type(password);
+    cy.get('input[name="username"]').type(username);
+    cy.get('input[name="password"]').type(password);
     cy.get('[type="submit"]').click();
+    cy.url().should('match', /\/admin\//);
+    cy.get('a[href*="/"]').contains('View site').click();
     cy.url().should('match', /\/tasks$/);
     cy.document().then((doc) => {
         const loadSettingFailNotice = Array.from(doc.querySelectorAll('.cvat-notification-notice-load-settings-fail'));
         loadSettingFailNotice.length > 0 ? cy.closeNotification('.cvat-notification-notice-load-settings-fail') : null;
     });
 });
 
-Cypress.Commands.add('logout', (username = Cypress.env('user')) => {
+Cypress.Commands.add('logout', () => {
     cy.get('.cvat-right-header').within(() => {
-        cy.get('.cvat-header-menu-dropdown').should('have.text', username).trigger('mouseover', { which: 1 });
+        cy.get('.cvat-header-menu-dropdown').trigger('mouseover', { which: 1 });
     });
     cy.get('span[aria-label="logout"]').click();
     cy.url().should('include', '/auth/login');
     cy.visit('/auth/login'); // clear query parameter "next"
     cy.closeModalUnsupportedPlatform();
 });
 
-Cypress.Commands.add('userRegistration', (firstName, lastName, userName, emailAddr, password) => {
-    cy.get('#firstName').type(firstName);
-    cy.get('#lastName').type(lastName);
-    cy.get('#username').type(userName);
-    cy.get('#email').type(emailAddr);
-    cy.get('#password1').type(password);
-    cy.get('#password2').type(password);
-    cy.get('.register-form-button').click();
+Cypress.Commands.add('userRegistration', (userName, emailAddr, walletAddr, signedEmail) => {
+    let authKey;
+    cy.request({
+        method: 'POST',
+        url: '/api/v1/auth/register',
+        body: {
+            username: userName,
+            email: emailAddr,
+            wallet_address: walletAddr,
+            signed_email: signedEmail,
+        },
+    }).then((response) => {
+        authKey = response['body']['key'];
+    });
+    cy.visit('/', {
+        headers: {
+            Authorization: `Token ${authKey}`,
+        },
+    });
```

Comment:
are you sure that this is working? the whole construction

`cy.request({}).then(() => {})` is asyncronus. But then you are doing `cy.visit(..)` which should wait for the `authKey` fulfillment. So I'm thinking that this `authKey` should be `undefined` at this moment

AND:

no need to create global var here. just push it to `then` pipeline:

```
cy.request({})
    .then(response => response['body']['key'])
    .then(token => cy.visit('/', {.. // expression with auth key}
```

Reply:
or cypress requests are synchronous?